### PR TITLE
Enable serviceMonitor support for all jaeger components

### LINKED ISSUE #96: [jaeger] Prometheus metrics
Hey!
I wonder why there is no ServiceMonitor for jaeger components? Is there any issues with it? If it's not, I'll happy to contribute this improvement. Let me know, what do you think about it =) 

I see list of ports and default path in docs: https://www.jaegertracing.io/docs/1.17/monitoring/

### PR DESCRIPTION
**Pull request by Jasstkn**
Hello! I've prepared changes for serviceMonitor. I would be glad if you can review it and leave some feedback. Let me know if there any problems.

resolves #96 

Signed-off-by: Jasstkn <jasssstkn@yahoo.com>

### CODE REVIEW

**Comment by Jasstkn** [comment]
P.S. I'm not sure if I can link this PR with this issue: https://github.com/jaegertracing/helm-charts/issues/96

**Comment by naseemkullah** [comment]
> P.S. I'm not sure if I can link this PR with this issue: #96

I think with one of these keywords it will work: https://help.github.com/en/enterprise/2.16/user/github/managing-your-work-on-github/closing-issues-using-keywords#about-issue-references


e.g.

closes #96 

🤞

File: charts/jaeger/values.yaml [hunk 1]
```diff
@@ -163,10 +163,14 @@ ingester:
...
 
+  serviceMonitor:
+    enabled: false
+    additionalLabels: {}
+
 agent:
   podSecurityContext: {}
   securityContext: {}
-  enabled: true
+  enabled: false
```
Comment (naseemkullah) [inline charts/jaeger/values.yaml:173]: This should not be changed, the default is to deploy an agent daemonset and is unrelated to security monitor.

**Comment by Jasstkn** [comment]
> > P.S. I'm not sure if I can link this PR with this issue: #96
> 
> I think with one of these keywords it will work: https://help.github.com/en/enterprise/2.16/user/github/managing-your-work-on-github/closing-issues-using-keywords#about-issue-references
> 
> e.g.
> 
> closes #96
> 
> 🤞

You're right =)

File: charts/jaeger/templates/query-servicemonitor.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      scrapeTimeout: {{ .Values.query.serviceMonitor.scrapeTimeout }}
+      {{- end }}
+  namespaceSelector:
+    matchNames:
+      - {{ .Release.Namespace }}
+  selector:
+    matchLabels:
+      app.kubernetes.io/component: query
+      app.kubernetes.io/instance: {{ .Release.Name }}
+{{- end }}
```
Comment (naseemkullah) [inline charts/jaeger/templates/query-servicemonitor.yaml:36]: nit: empty line missing at eof

File: charts/jaeger/templates/ingester-servicemonitor.yaml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+      scrapeTimeout: {{ .Values.ingester.serviceMonitor.scrapeTimeout }}
+      {{- end }}
+  namespaceSelector:
+    matchNames:
+      - {{ .Release.Namespace }}
+  selector:
+    matchLabels:
+      app.kubernetes.io/component: ingester
+      app.kubernetes.io/instance: {{ .Release.Name }}
+{{- end }}
```
Comment (naseemkullah) [inline charts/jaeger/templates/ingester-servicemonitor.yaml:36]: nit: empty line missing at eof

File: charts/jaeger/templates/collector-servicemonitor.yaml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+      scrapeTimeout: {{ .Values.collector.serviceMonitor.scrapeTimeout }}
+      {{- end }}
+  namespaceSelector:
+    matchNames:
+      - {{ .Release.Namespace }}
+  selector:
+    matchLabels:
+      app.kubernetes.io/component: collector
+      app.kubernetes.io/instance: {{ .Release.Name }}
+{{- end }}
```
Comment (naseemkullah) [inline charts/jaeger/templates/collector-servicemonitor.yaml:36]: nit: empty line missing at eof

File: charts/jaeger/templates/agent-servicemonitor.yaml [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+      scrapeTimeout: {{ .Values.agent.serviceMonitor.scrapeTimeout }}
+      {{- end }}
+  namespaceSelector:
+    matchNames:
+      - {{ .Release.Namespace }}
+  selector:
+    matchLabels:
+      app.kubernetes.io/component: agent
+      app.kubernetes.io/instance: {{ .Release.Name }}
+{{- end }}
```
Comment (naseemkullah) [inline charts/jaeger/templates/agent-servicemonitor.yaml:36]: nit: empty line missing at eof

**Comment by naseemkullah** [comment]
Thanks @Jasstkn !

**Pull request merged**